tests/hypothesis_strategies/priors.py

```diff
@@ -11,9 +20,54 @@
 )
 """A strategy that generates Gamma priors."""
 
+half_cauchy_priors = st.builds(
+    HalfCauchyPrior,
+    st.floats(min_value=0, exclude_min=True),
+)
+"""A strategy that generates Half-Cauchy priors."""
+
+normal_priors = st.builds(
+    NormalPrior,
+    st.floats(allow_nan=False, allow_infinity=False),
+    st.floats(min_value=0, exclude_min=True),
```

Comment:
I was wondering the same, seems like `st.floats` only considers nans and infinities if min and max are none so its necessary to exclude them only in the unbounded case

Reply:
Hm, are we sure? Wondering that this never bothered me earlier ...

Reply:
here for `allow_nan`
![image](https://github.com/emdgroup/baybe/assets/17951239/ce3b6b05-e6f4-4de5-866a-571eb6f2bb89)

theres another reason why it must be true: hypothesis configured in this manner does not create breaking code/priors

Reply:
Ok, makes sense for `nan`, but for `infinity`, the logic is actually different, as far as I can see!? Looks like `inf` should still be generated with the current setting, or am I mistaken?

Reply:
ok true that, seems this has slipped through because theres no iteration test with hypothesis-generated priors and I guess the serialization tests were still working? I guess theres no chance it simply doesnt choose inf?

Reply:
do you think I should adjust this now? Is there a chance this also has to be adjusted in many other files? I guess this would anyway unearth and be easily fixed once we enable iteration tests or similar with hypothesis generated objects

Reply:
Oh, just saw your message after having sent mine... But have a look at my draft =)